nifi-asymmetrik-standard-bundle/nifi-asymmetrik-standard-processors/src/main/java/com/asymmetrik/nifi/processors/s3/ListS3Batch.java

```diff
@@ -0,0 +1,361 @@
+package com.asymmetrik.nifi.processors.s3;
```

Comment:
The general motivation here is (at least) two things:

1) The ListS3 processor chokes up NiFi if the number of objects is large, and there's no way to configure that processor to list objects incrementally. This processor can be so configured.
2) The ListS3 processor does not allow listing files falling between two prefixes, so, for example, given a set of objects stored by date, it's not possible (without altering the processor 18 times), list the past 18 months worth of objects. This processor allows the specification of a start and end S3 prefix.

Reply:
We try hard to determine the value of promoting project-related processors to corporate NARS.  When the value is there, we have to go the extra mile to ensure they are written for general usage outside of Asymmetrik.  I think this one is worth promoting.  That being said there are more requirements to make this work well within the large NiFi community.  Here are a few critical features currently missing:

1. There needs to be a way to specify an authentication provider
1. There needs to be an endpoint override URL
1. There needs to be support for an SSL context
1. There needs to be support to write the object tags

Reply:
For #1, we can add credentials file, access key and secret key - like the CloudwatchNiFiClusterMetricsReporter. Adding a AWSCredentialsProviderService is problematic because of how it is packaged by NiFi.

Fair points on the others items. We'll try to align the properties and output attributes to ListS3 as best as possible.

Reply:
I can live without the credential provider service as long as there's the credential file and access / secret key options.  For context, there is a network that contains S3 compliant storage, but it needs to authenticate with access key and secret key, and it needs an endpoint override.